### DISCUSSION TITLE
Custom commands not working as expected

### DISCUSSION THREAD

**Discussion by RaeedAhmed** [starship/starship#3074] [Q&A]
As a custom prompt I want the parent directory as the name for an active python virtualenv.
As an example, say i have a project with a root directory called `myproject` and i have a virtual environment folder `.venv` that is located in the root directory. 

The default prompt behavior is to show the name `.venv` as that is the environment folder name, but i want `myproject` instead.

The following command outputs `myproject` from anywhere as it is using the shell's `VIRTUAL_ENV` variable:

`echo $(basename $(dirname $VIRTUAL_ENV))`

In `.config/starship.toml`, I have the following custom command:

```
[custom.venv]
style = "dimmed cyan"
command = "echo $(basename $(dirname $VIRTUAL_ENV))"
format = "[$output]($style)"
when = """ test $VIRTUAL_ENV """
```

This does not seem to print anything in the prompt. However, if i change the command to `"echo $(basename $VIRTUAL_ENV)"`, the output `.venv` does show up.

I have also tried creating a function in my `.bashrc`:

```
venvname ()
{
	basename $(dirname "$VIRTUAL_ENV")
}
```

and replacing the command to "venvname".

I'm not sure what I'm doing wrong. I am also open to suggestions on a different approach to what I've been trying.

**Comment by simonvanderveldt** [comment]
@RaeedAhmed FYI your example is working for me using starship 1.12.0

I was looking to do the same thing as you, show the parent directory that contains the `.venv` directory instead of just `.venv` since that isn't very meaningful. Might be a common enough use-case to request to add it to python module?
